R/object_usage_linter.R

```diff
@@ -29,7 +35,23 @@
 #' @evalRd rd_linters("package_development")
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
-object_usage_linter <- function(interpret_glue = TRUE, skip_with = TRUE) {
+object_usage_linter <- function(interpret_glue = NULL, interpret_extensions = c("glue", "rlang"), skip_with = TRUE) {
+  if (!is.null(interpret_glue)) {
+    lintr_deprecated(
+      "interpret_glue",
+      '"glue" in interpret_extensions',
+      version = "3.3.0",
+      type = "Argument",
+      signal = "warning"
+    )
+
+    interpret_extensions <- (if (interpret_glue) union else setdiff)(interpret_extensions, "glue")
```

Comment:
That's golfy ^^
Since `TRUE` is a default here, I think we can get away with

```suggestion
    if (!interpret_glue) {
      interpret_extensions <- setdiff(interpret_extensions, "glue")
    }
```

WDYT?

Reply:
That doesn't cover `object_usage_linter(interpret_extensions = "rlang")` right? We'd have to ignore the previous default for `interpret_glue`.

That may be fine, since using `interpret_extensions=` _should_ imply that you've already adapted to the new argument. But the symmetry I opted for is easier to reason about. We can just "uncrack" the conciseness a bit.

Reply:
Doesn't the current approach also force setting `interpret_glue = FALSE`?

`if (!mssing(interpret_glue) && !interpret_glue) ...`
would be a working condition, right?

Reply:
PTAL. I hear you, but the resulting code is IMO harder to read, I think right now it's very transparent.